### DISCUSSION TITLE
Why there is a hash sign in MLFLow URL?

### DISCUSSION THREAD

**Discussion by pkubik** [mlflow/mlflow#6088] [Q&A] [answered]
All MLFlow UI resources seem to start with hash sign, e.g. the home page is `http://127.0.0.1:5000/#/`. Out of curiosity, what does it mean? I didn't see this in any other web service but I'm not very knowledgeable in web technologies. Obviously, typically `#` stands for an anchor, but those usually occur at the and of the URL, not at the beginning of the resource string.

I was wondering whether it is some nice hack to make it work smoother with reverse proxies? Any ideas?

**Comment by dbczumar** [comment, marked as answer]
Hi @pkubik. Thank you for your question. The hash is present due to MLflow's use of the HashRouter package: https://v5.reactrouter.com/web/api/HashRouter, which is leveraged here: https://github.com/mlflow/mlflow/blob/3d5777642b67477414f33de258d3394c81909344/mlflow/server/js/src/experiment-tracking/components/App.js#L5. This helps certain third parties such as Databricks, serve the MLflow UI. Thank you for using MLflow!
